### DISCUSSION TITLE
Opinion on changing the license regime for beckn protocol specification

### DISCUSSION THREAD

**Discussion by ravi-prakash-v** [beckn/protocol-specifications#259] [Proposals]
With the increased adoption of the protocol, there are innovations flourishing around the ecosystem. To accommodate all those, there are dynamic needs or change recommendations that need to be adopted by the specification. Sometimes the sheer volume of such changes may outpace the governance or sometimes the change requests are just not aligned with the major roadmap for Beckn protocol. In such a situation a Non-derivative (CC-BY-ND) license can occur to be a major roadblock to such adoptions. Keeping those scenarios in mind, the proposal hereby is being raised to make the licensing form a bit more lenient and easier to adopt.

The proposed alternative hereby is a CC-BY-NC-SA 4.0 license. Some salient features of the same is as mentioned below-

- As per section 2, this license will give users a worldwide, royalty-free, non-sublicensable, non-exclusive, irrevocable license to (i) reproduce and share the licensed works i.e. the specifications, and (ii) to produce, reproduce and share modified versions of the work/ specifications, both for non-commercial purposes. 

- In case of sharing, as per section 3(a), the following must be retained to ensure compliance with attribution requirements: the specifications must retain identification of creators, a copyright notice, a notice that refers to the CC-BY-SA-NC license (ULR), a notice that refers to the disclaimer of warranties (as listed in section 5) and a ULR to the licensing specifications. 

- Users must indicate if they have modified the specifications and retain an indication of any previous modifications. As per section 3(b), if the original specifications have been modified, the modified work must be distributed under the same license. 

Since this license does not restrict modifications and sharing of modifications, contextualized adoption is not restricted. To the benefit of the community, any future user cannot add new restrictions to modified / derivate versions of the specifications since their modified versions must also be licensed in the same way and under 'share-alike' conditions. The license also requires users to indicate the modifications that they have made thereby enabling the governing body to track modifications. This license will also restrict commercial uses of the specifications which will limit the possibility of users using the specifications for private, commercial gain. The attribution requirements will ensure that original creators are given due credit. 

Therefore, it is proposed to move our current license regime from CC-BY-ND to CC-BY-NC-SA to encourage Cambrian explosion in terms of adoption as well as robust governance to track and handle the changes.

**Comment by ravi-prakash-v** [comment]
Requesting the @beckn/core-working-group to comment on the same.

**Reply by venkatramanm** [reply]
I am fine with the proposal

**Reply by pkapustin** [reply]
NC clause will still allow building commercial products that use the protocol or its derivatives right?
In this case sounds great.

**Reply by pramodkvarma** [reply]
Yes @pkapustin 

@core-wg-admin I am fine with the proposal.

**Comment by sjthnrk** [comment]
I am fine with the proposal. 

As a working group and as members of larger community, we must strive to recognise the design principles and interoperability as an integral part of the evolution of beckn protocol,  even through the derivative works.

**Reply by pramodkvarma** [reply]
Absolutely @sjthnrk! Can't agree more!

**Comment by ravi-prakash-v** [comment]
Great! As we have consensus. I shall raise a PR containing the new license to the `governance-draft` branch that will ultimately be merged to `governance-main` by way of PR review. 

`governance-main` will be synced with `master` when the next version of protocol is released.

**Comment by ravi-prakash-v** [comment]
Raised a PR to `governance-main` from `governance-draft` : https://github.com/beckn/protocol-specifications/pull/262

Requesting you all to approve changes. Thanks
